compiler/noirc_evaluator/src/ssa/interpreter/mod.rs

```diff
@@ -278,6 +278,34 @@ impl<'ssa> Interpreter<'ssa> {
         self.lookup_helper(value_id, instruction, "array or slice", Value::as_array_or_slice)
     }
 
+    fn lookup_bytes(&self, value_id: ValueId, instruction: &'static str) -> IResult<Vec<u8>> {
+        let array =
+            self.lookup_helper(value_id, instruction, "array or slice", Value::as_array_or_slice)?;
+        let array = array.elements.borrow();
+        let mut error = None;
+        let result = array
+            .iter()
+            .map(|v| {
+                if let Some(v) = v.as_u8() {
+                    v
+                } else {
+                    error = Some(v);
+                    0
+                }
+            })
+            .collect::<Vec<u8>>();
+        if let Some(v) = error {
+            Err(internal(InternalError::TypeError {
+                value_id,
+                value: v.to_string(),
+                expected_type: "u8",
+                instruction,
+            }))
+        } else {
+            Ok(result)
+        }
```

Comment:
Can collect to a Result directly:
```suggestion
        let result = array
            .iter()
            .map(|v| v.as_u8().ok_or_else(|| {
                internal(InternalError::TypeError {
                    value_id,
                    value: v.to_string(),
                    expected_type: "u8",
                    instruction,
                }
            }))
            .collect::<Result<Vec<u8>>, _>()
```
The type hint on collect should now be unnecessary since its being directly returned but I'm showing it because in general you often need the hint when collecting to a `Result<Vec<_>, _>`